_src/pattern-library/sass/components/_forms.scss

```diff
@@ -0,0 +1,425 @@
+// #SELECT
+// ----------------------------
+.wrapper-replace-select {
+    position: relative;
+}
+
+.input-select {
+    height: ($baseline*2) + px;
+    line-height: ($baseline*2) + px;
+
+    &:focus,
+    &.is-active {
+        border-color: $field-default-border-focus;
+        color: $field-default-text-color-hover;
+    }
+
+    &:disabled,
+    &.is-disabled {
+        border-color: $field-disabled-border;
+        background: $field-disabled-background;
+    }
+
+    &.is-replaced {
+        position: absolute;
+        z-index: z-index(mid-front);
+    }
+
+    &.is-transparent {
+        opacity: 0;
+    }
+}
+
+.wrapper-custom-select {
+    @extend .input;
+    @extend .input-select;
+    position: absolute;
+    display: inline;
+    top: 0;
+    left: 0;
+    width: auto;
+    line-height: 100%;
+    z-index: z-index(base);
+
+    svg {
+        position: absolute;
+        right: ($baseline/2) + px;
+        height: $baseline + px;
+        width: $baseline + px;
+        fill: $field-default-border;
+    }
+
+    &:hover,
+    &:focus,
+    &.is-hover,
+    &.is-active {
+        border: 1px solid $field-default-border-focus;
+
+        svg {
+            @include transition(all timing(xx-fast) ease-in-out 0s);
+            fill: $field-default-border-focus;
+        }
+    }
+
+    &:disabled,
+    &.is-disabled {
+        cursor: default;
+        color: $field-disabled-text-color;
+    }
+}
+
+// ----------------------------
+// #RADIO / CHECKBOX
+// ----------------------------
+.wrapper-replace-radio,
+.wrapper-replace-checkbox {
+    position: relative;
+    display: inline-block;
+    width: auto;
+    vertical-align: bottom;
+}
+
+.wrapper-custom-radio,
+.wrapper-custom-checkbox {
+    display: inline-block;
+    z-index: z-index(base);
+
+    svg {
+        @include transition(all timing(xx-fast) ease-in-out 0s);
+        height: $baseline + px;
+        width: $baseline + px;
+        fill: $field-default-border;
+    }
+
+    &.is-selected {
+
+        svg {
+            fill: $field-default-border-focus;
+        }
+    }
+}
+
+.input-radio,
+.input-checkbox {
+
+    &:disabled,
+    &.is-disabled {
+
+        svg {
+            fill: $field-disabled-text-color;
+            cursor: default;
+        }
+    }
+
+    &.is-replaced {
+        position: absolute;
+        margin-top: ($baseline/4) + px;
+        z-index: z-index(mid-front);
+    }
+
+    &.is-transparent {
+        opacity: 0;
+    }
+}
+
+// ----------------------------
+// #TEXTAREA
+// ----------------------------
+.input-textarea {
+
+    &:focus,
+    &:active,
+    &.is-active {
+        border-color: $field-default-border-focus;
+        color: $field-default-text-color-hover;
+    }
+
+    &.textarea-short {
+        height: ($baseline*4.5) + px; // 3 clean lines
+    }
+
+    &.textarea-medium {
+        height: ($baseline*10) + px; // 8 clean lines
+    }
+
+    &.textarea-tall {
+        height: ($baseline*14.4) + px; // 12 clean lines
```

Comment:
14.4 is crazy value and seems to be customized to scale well with our ``$baseline``, but I can't say it will with any overrides of that variable. Please make this more bulletproof for general resetting of ``$baseline`` by creating a scale for short -> tall.